### PR TITLE
Threshold parameter validation

### PR DESCRIPTION
**Pull request by agouin**
Add assertions for t > n/2 for both key sharding and cosigner daemon start.

https://github.com/strangelove-ventures/horcrux/blob/main/docs/signing.md#private-key-sharding-and-number-of-nodes

### CODE REVIEW

**Review by mark-rushakoff — APPROVED** [review]
Prefer `t.Setenv` in tests. One question about `<` vs `<=`, otherwise good to ogo.

File: cmd/horcrux/cmd/config_test.go [hunk 1]
```diff
@@ -317,18 +296,10 @@ func TestConfigNodesAddAndRemove(t *testing.T) {
...
-	})
 }
 
 func TestConfigPeersAddAndRemove(t *testing.T) {
-	tmpHome := "/tmp/TestConfigPeersAddAndRemove"
-
-	err := os.Setenv("HOME", tmpHome)
-	require.NoError(t, err)
-	err = os.MkdirAll(tmpHome, 0777)
+	err := os.Setenv("HOME", t.TempDir())
```
[the remark this answers is not in the document]
Comment (DavidNix) [reply cmd/horcrux/cmd/config_test.go:302]: TIL 🎉

File: cmd/horcrux/cmd/key2shares.go [hunk 2]
```diff
@@ -66,11 +66,20 @@ func validateCreateCosignerShares(cmd *cobra.Command, args []string) error {
...
+	t, err := strconv.ParseInt(args[1], 10, 64)
+	if err != nil {
+		return fmt.Errorf("threshold must be an integer, got(%s)", args[1])
 	}
-	if _, err := strconv.ParseInt(args[2], 10, 64); err != nil {
-		return fmt.Errorf("threshold must be an integer got(%s)", args[2])
+	n, err := strconv.ParseInt(args[2], 10, 64)
+	if err != nil {
+		return fmt.Errorf("shares must be an integer, got(%s)", args[2])
+	}
```
[the remark this answers is not in the document]
Comment (DavidNix) [reply cmd/horcrux/cmd/key2shares.go:76]: Agreed. My preferred default is always wrapping the error (never swallow it). The only exception is if the error bubbles up to the end user somehow, exposing sensitive info, and could be considered a security issue.

File: cmd/horcrux/cmd/config_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (agouin) [reply cmd/horcrux/cmd/config_test.go]: done

File: cmd/horcrux/cmd/key2shares.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (agouin) [reply cmd/horcrux/cmd/key2shares.go]: Good call, updated.

**Pull request merged**